Lib/test/test_dbm.py

```diff
@@ -129,6 +130,9 @@ def test_anydbm_access(self):
         assert(f[key] == b"Python:")
         f.close()
 
+    def test_open_with_pathlib_path(self):
+        dbm.open(Path(_fname), "c").close()
```

Comment:
Use `os.support.os_helper.FakePath` for testing.

Test also with bytes path after adding support of it.